### ISSUE TITLE
API: HTML Documentation for Latest Version of Elektra Is Missing

### ISSUE THREAD

**Issue by sanssecours** [ElektraInitiative/libelektra#2673]
# Steps to Reproduce the Problem

Open the URL: https://doc.libelektra.org/api/latest/html

## Expected Result

The browser opens a page that contains the latest version of the Doxygen documentation generated from the `master` branch. This document should look similar to the Doxygen documentation for the latest **released** version of Elektra located [here](https://doc.libelektra.org/api/current/html).

## Actual Result

The link above returns a “404 Not Found” HTML page.

**Comment by markus2330** [comment]
Thank you for reporting this important bug.

From the https://github.com/ElektraInitiative/doc/ repository it looks like that "current" is the "current" release and latest should be the docu of the master branch.

And scripts/jenkins/Jenkinsfile also tries to write to "latest". For some reason, however, it seems to be not working. Maybe it was just because master/some PR failed to update the folder in the last build run (the last build seems to have failed at very late point)? master is currently building again, let us see if the files are still not there afterwards. (Even if there are it is still a bug that docu disappears for so long.)

**Comment by markus2330** [comment]
It seems to be a permanent issue, it seems like the sshPublisher in scripts/jenkins/Jenkinsfile line 745 to 772 does not work or does not find files. The output from https://build.libelektra.org/jenkins/blue/organizations/jenkins/libelektra/detail/master/713/pipeline/416 is:

```
SSH: Connecting from host [i7]
SSH: Connecting with configuration [doc.libelektra.org] ...
SSH: Disconnecting configuration [doc.libelektra.org] ...
SSH: Transferred 1704 ( 1704 + 0 + 0 + 0 ) file(s)
```

I tried to simply create an empty html+man folder on the server:

```sh
cd /srv/libelektra/api/latest
mkdir man
mkdir html
chown jenkins html man
```

**Comment by markus2330** [comment]
The html pages are maybe simply not copied because "build/doc/html/*" is missing in the Jenkinsfile, I wonder how this ever worked? But I still do not know why the man pages are missing, the glob seems to be correct?

Furthermore the specification "doc/help/*.html'" seems to be wrong ("build" missing? Or should this refer to doc/man/man*?) This might have happened due to restructure in Elektra's source.

@ingwinlu can you shine some light on how this worked? (copy of html api docu)

**Comment by ingwinlu** [comment]
I don't think any of this was done automatically ever as the old doc build job also had no notion of doing this.

**Comment by markus2330** [comment]
@ingwinlu thank you very much, this clarifies everything!

So it seems like we need to extend the Jenkinsfile to copy the docu (it is already build). I'll take the issue.

**Comment by markus2330** [comment]
Is hopefully fixed with  #2737

**Comment by markus2330** [comment]
Link still seem to work.
